scipy/optimize/_bracket.py

```diff
@@ -365,11 +374,11 @@ def customize_result(res, shape):
         res['fl'] = fl
         res['fr'] = fr
 
-        res['nit'] = np.maximum(res['nit'][:n], res['nit'][n:])
+        res['nit'] = xp.maximum(res['nit'][:n], res['nit'][n:])
         res['nfev'] = res['nfev'][:n] + res['nfev'][n:]
         # If the status on one side is zero, the status is zero. In any case,
         # report the status from one side only.
-        res['status'] = np.choose(sa == 0, (sb, sa))
+        res['status'] = xp.where(sa != 0, sb, sa)
```

Comment:
```suggestion
        res['status'] = xp.where(sa == 0, sa, sb)
```
I'll try this locally and commit if it works. Unless I'm going myself, it's easier for me to compare with the comment.